app/abr-menu.js

```diff
@@ -197,6 +199,50 @@ AbrMenu.prototype = {
     attach: function () {
         // FIXME: use win.setMenu(menu) for Linux and Windows instead
         Menu.setApplicationMenu(this.menu);
+    },
+
+    openRecentDoc: function (recentFile) {
+        this.abrWin.abrApp.open(recentFile);
+        // FIXCC recent-docs: Recent doc list is not updated after re-opening a recent file
+    },
+
+    clearRecentDocs: function () {
+        this.abrWin.abrApp.clearRecentDocs(this.abrWin);
+    },
+
+    setRecentDocsMenu: function(recentPaths) {
+        var submenu = this.findItem("recentDocs").submenu, i, itemData, that = this;
+
+        // clear menu (this API is not public and may change between Electron releases)
+        submenu.clear();
+
+        function createOpenRecentCallback(recent) {
+            return function() {
+                that.openRecentDoc(recent);
+            };
+        }
+
+        // create items for recent files
+        for (i=0; i <recentPaths.length; i++) {
+            itemData = {
+                label: recentPaths[i],
+                click: createOpenRecentCallback(recentPaths[i])
+            };
+            submenu.append(new MenuItem(itemData));
+        }
+
+        submenu.append(new MenuItem({
+            type: "separator"
+        }));
+
+        // append "clear recent" item
+        submenu.append(new MenuItem({
+            label: "Clear",
```

Comment:
Here, I could use the "i18n" branch (still to me merged "develop") to i18nize the "Clear" menu label.